usr/lib/common/asn1.c

```diff
@@ -411,8 +411,109 @@ CK_RV ber_decode_OCTET_STRING(CK_BYTE *str,
     return CKR_FUNCTION_FAILED;
 }
 
-//
-//
+/**
+ * Here we assume that the indication about unused bits is NOT
+ * part of the data. It will be added by this function.
+ */
+CK_ULONG ber_encode_BIT_STRING(CK_BBOOL length_only,
+                            CK_BYTE **ber_str,
```

Comment:
do we actually need length_only param ? can we not use the standard "if outbuf == NULL then return outbuflen" ?

Reply:
Yes, I think we could. Let's put this on our list of housekeeping todo's. I counted 16 routines in asn1.c that are coded this way.